docs/source/en/conceptual/reproducibility.mdx

```diff
@@ -0,0 +1,159 @@
+
+```python
+from diffusers import DDIMPipeline
+import numpy as np
+
+model_id = "google/ddpm-cifar10-32"
+
+# load model and scheduler
+ddim = DDIMPipeline.from_pretrained(model_id)
+
+# run pipeline for just two steps and return numpy tensor
+image = ddim(num_inference_steps=2, output_type="np").images
+print(np.abs(image).sum())
+```
+
+Running the above prints a value of 1464.2076, but running it again prints a different 
+value of 1495.1768. What is going on here? Every time the pipeline is run, gaussian noise 
+is created and step-wise denoised. To create the gaussian noise with [`torch.randn`](https://pytorch.org/docs/stable/generated/torch.randn.html) a different random seed is taken every time, thus leading to a different result.
+This is a desired property of diffusion pipelines as it means that the pipeline can create a different random image every time it is run. In many cases, one would like to generate the exact same image of a certain 
+run, for which case an instance of a [PyTorch generator](https://pytorch.org/docs/stable/generated/torch.randn.html) has to be passed:
+
+```python
+import torch
+from diffusers import DDIMPipeline
+import numpy as np
+
+model_id = "google/ddpm-cifar10-32"
+
+# load model and scheduler
+ddim = DDIMPipeline.from_pretrained(model_id)
+
+# create a generator for reproducibility
+generator = torch.Generator(device="cpu").manual_seed(0)
+
+# run pipeline for just two steps and return numpy tensor
+image = ddim(num_inference_steps=2, output_type="np", generator=generator).images
+print(np.abs(image).sum())
+```
+
+Running the above always prints a value of 1491.1711 - also upon running it again because we 
+define the generator object to be passed to all random functions of the pipeline.
+
+If you run this code snippet on your specific hardware and version, you should get a very similar, if not the same result.
+
+<Tip>
+
+It might be a bit unintuitive at first to pass "generator" objects to the pipelines instead of 
+just integer values representing the seed, but this is the recommended design when dealing with 
+probabilistic models in PyTorch as generators are *random states* that are advanced and can thus be 
+passed to multiple pipelines in a sequence.
+
+</Tip>
+
+Great! Now, we know how to write reproducible pipelines, but it does get a bit tricker since the above example only runs on CPU. How do we also achieve reproducibility on GPU? 
+In short, one should not expect to get full reproducibility across hardwares when running pipelines on GPU 
+as matrix multiplications are less deterministic on GPU than on CPU and diffusion pipelines tend to require
+a lot of matrix multiplications. Let's see what we can do though to keep the randomness within limits across 
+different GPU hardware.
+
+To achieve maximum speed performance, it is recommended to create the generator directly on GPU when running 
+the pipeline on GPU:
```

Comment:
```suggestion
<Tip>

It might be a bit unintuitive at first to pass `generator` objects to the pipelines instead of 
just integer values representing the seed, but this is the recommended design when dealing with 
probabilistic models in PyTorch as generators are *random states* that are advanced and can thus be 
passed to multiple pipelines in a sequence.

</Tip>

Great! Now, we know how to write reproducible pipelines, but it gets a bit trickier since the above example only runs on the CPU. How do we also achieve reproducibility on GPU? 
In short, one should not expect full reproducibility across different hardware when running pipelines on GPU 
as matrix multiplications are less deterministic on GPU than on CPU and diffusion pipelines tend to require
a lot of matrix multiplications. Let's see what we can do to keep the randomness within limits across 
different GPU hardware.

To achieve maximum speed performance, it is recommended to create the generator directly on GPU when running 
the pipeline on GPU:
```